src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -285,5 +302,73 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
             result = results;
             return true;
         }
+
+        internal static bool TryReadStatusFile(int pid, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            bool b = TryParseStatusFile(GetStatusFilePathForProcess(pid), out result, reusableReader);
+            Debug.Assert(!b || result.pid == pid, "Expected process ID from status file to match supplied pid");
+            return b;
+        }
+
+        private static bool TryParseStatusFile(string statusFilePath, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            string statusFileContents;
+            try
+            {
+                using (var source = new FileStream(statusFilePath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false))
+                {
+                    statusFileContents = reusableReader.ReadAllText(source);
+                }
+            }
+            catch (IOException)
+            {
+                // Between the time that we get an ID and the time that we try to read the associated status
+                // file(s), the process could be gone.
+                result = default(ParsedStatus);
+                return false;
+            }
+
+            var parser = new StringParser(statusFileContents, '\n');
+            var results = default(ParsedStatus);
+
+            var splits = new [] {':', ' ', '\t'};
+
+            while (parser.MoveNext())
+            {
+                var elements = parser.ExtractCurrent().Split (splits, 3, StringSplitOptions.RemoveEmptyEntries);
+                if (elements.Length < 2)
+                {
+                    continue;
+                }
+
+                switch (elements [0])
+                {
+                    case "Pid":
+                        results.pid = int.Parse (elements[1]);
+                        break;
+                    case "VmHWM":
+                        results.vmhwm = ulong.Parse (elements[1]) * 1024;
+                        break;
+                    case "VmRSS":
+                        results.vmrss = ulong.Parse (elements[1]) * 1024;
+                        break;
+                    case "VmData":
+                        results.vmdata = ulong.Parse (elements[1]) * 1024;
+                        break;
+                    case "VmSwap":
+                        results.vmswap = ulong.Parse (elements[1]) * 1024;
+                        break;
+                    case "VmSize":
+                        results.vmsize = ulong.Parse (elements[1]) * 1024;
+                        break;
+                    case "VmPeak":
+                        results.vmpeak = ulong.Parse (elements[1]) * 1024;
```

Comment:
Any chance these could overflow?  Should we cap the values to ulong.MaxValue?

Reply:
This would overflow if you can allocate more than 16384 PB of virtual memory. This is also limited by the fact we have 64bits system which can't allocate more than that amount of memory.

Reply:
>This would overflow if you can allocate more than 16384 PB of virtual memory.

I'm wondering if the system might ever output ulong.MaxValue as a sentinel, as it appears to for things like max working set.

Reply:
In the `status` file, the value is suffixed by `kB` strongly indicating the value should not overflow `ulong.MaxValue / 1024`. I'll add some protection around it, just to be sure.